src/main/java/Nell.java

```diff
@@ -0,0 +1,217 @@
+    }
+
+    /**
+     * Executes a deadline command with a given command body
+     *
+     * @param commandBody The command body
+     */
+    private static void executeCommandDeadline(String commandBody) {
+        System.out.println("-> The task has been added to the list:");
+        String[] commandWords = commandBody.split("/by");
+        Deadline deadlineToAdd = new Deadline(commandWords[0].trim(), commandWords[1].trim());
+        listAddTask(deadlineToAdd);
+        System.out.println("   " + deadlineToAdd);
+        System.out.println(String.format("   The list now has %d tasks", taskCount));
+    }
+
+    /**
+     * Executes an event command with a given command body
+     *
+     * @param commandBody The command body
+     */
+    private static void executeCommandEvent(String commandBody) {
+        System.out.println("-> The task has been added to the list:");
+        String[] commandWords = commandBody.split("/from|/to", 3);
+        Event eventToAdd = new Event(commandWords[0].trim(), commandWords[1].trim(), commandWords[2].trim());
+        listAddTask(eventToAdd);
+        System.out.println("   " + eventToAdd);
+        System.out.println(String.format("   The list now has %d tasks", taskCount));
+    }
+
+    /**
+     * Executes an unmark command with a given command body
+     *
+     * @param commandBody The command body
+     */
+    private static void executeCommandUnmark(String commandBody) {
+        int taskIndex = Integer.parseInt(commandBody);
+        if (checkIfTaskValid(taskIndex)) {
+            // Marks a task as not done
+            tasks[taskIndex - 1].setDone(false);
+            System.out.println("-> The following task has been marked not done:");
+            printTaskAtIndex(tasks[taskIndex - 1], taskIndex);
+        } else {
+            System.out.println("-> Invalid task!");
+        }
+    }
+
+    /**
+     * Executes a mark command with a given command body
+     *
+     * @param commandBody The command body
+     */
+    private static void executeCommandMark(String commandBody) {
+        int taskIndex = Integer.parseInt(commandBody);
+        if (checkIfTaskValid(taskIndex)) {
+            // Marks a task as done
+            tasks[taskIndex - 1].setDone(true);
+            System.out.println("-> The following task has been marked done:");
+            printTaskAtIndex(tasks[taskIndex - 1], taskIndex);
+        } else {
+            System.out.println("-> Invalid task!");
+        }
+    }
+
+    /**
+     * Executes an add task command (no command word) with a given command body
+     *
+     * @param command The command body
+     */
+    private static void executeCommandAddTask(String command) {
+        Task taskToAdd = new Task(command);
+        listAddTask(taskToAdd);
+        System.out.printf("-> added: %s%n", command);
+    }
+
+    /**
+     * Executes a bye command
+     *
+     */
+    private static void executeCommandBye() {
```

Comment:
good naming convention, however I would suggest you to remove "executeCommand" from your function names (ie just leaving the command name as the function name). This way it improves readability of the code

Reply:
This makes sense, but for some of these commands (like "todo" and "deadline"), I believe that following your suggestion would make the method name potentially misleading, and also go against naming conventions (method names being verbs). Would it be better for me to change the method names to "execute<command>" instead?